Library/Homebrew/version.rb

```diff
@@ -171,6 +171,41 @@ def detected_from_url?
     end
   end
 
+  class NullVersion
```

Comment:
Since this is effectively a singleton class, wouldn't it make more sense to just do `NULL = Version.new do …`? That way, instead of checking `version.is_a?(Version::NullVersion)` or `version > "0"`, you could just check `version.equal?(Version::NULL)`